invenio_deposit/api.py

```diff
@@ -243,7 +243,10 @@ def create(cls, data, id_=None):
 
             data['_deposit']['created_by'] = creator_id
 
-        return super(Deposit, cls).create(data, id_=id_)
+        deposit = super(Deposit, cls).create(data, id_=id_)
+        # FIXME create the bucket!
+        deposit.files.bucket
```

Comment:
Maybe in the `FilesMixin`?
The problem appeared when I tried to use the "vanilla" `/deposit/new` to create a new deposit.
These are the steps followed by the `UI`:
 1. Create the deposit with POST
 2. Read from the links the bucket endpoint
 3. Push the file to the bucket endpoint with PUT.

Without this "hack", the [bucket is created](https://github.com/inveniosoftware/invenio-records-files/blob/master/invenio_records_files/api.py#L291) when the deposit is serialized in JSON, i.e. after the session commit().
It means that the bucket doesn't exist anymore when the file is uploaded.

Maybe we should force the creation on [deposit creation](https://github.com/inveniosoftware/invenio-records-files/blob/master/invenio_records_files/api.py#L250)?
Something like:
```python
class FilesMixin(object):
   def create(self, *args, **kwargs):
      self._create_bucket()
      super(FilesMixin, self).create(*args, **kwargs)
```